src/pytorch_lightning/loggers/mlflow.py

```diff
@@ -250,7 +260,7 @@ def finalize(self, status: str = "FINISHED") -> None:
             self.experiment.set_terminated(self.run_id, status)
 
     @property
-    def save_dir(self) -> Optional[str]:
+    def save_dir(self) -> str:
```

Comment:
Why was this changed? It returns `None` if `not self._tracking_uri.startswith(LOCAL_FILE_URI_PREFIX)`

Reply:
self._tracking_uri gets set to a str on init if not provided by the user, with the following at ln 138:

```sh
if not tracking_uri:
    tracking_uri = f"{LOCAL_FILE_URI_PREFIX}{save_dir}"
```

tracking_uri is optional, but it's seem that save_dir can always expect to return a str

Reply:
But if a user creates a `MLFlowLogger(tracking_uri="foobar")` then the `startswith` will resolve to False and `save_dir` will return `None`. It expects to return a `str` if `tracking_uri` was not provided or was provided as expected.

Reply:
and the arg save_dir is set to default str; should this have been None?

`save_dir: Optional[str] = "./mlruns"`